ztp/siteconfig-generator/siteConfig/siteConfigHelper.go

```diff
@@ -0,0 +1,129 @@
+package siteConfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	mcfgctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+
+	"github.com/lack/yamltrim"
+	yaml "gopkg.in/yaml.v3"
+)
+
+const mcRoleKey = "machineconfiguration.openshift.io/role"
+const mcName = "predefined-extra-manifests"
+
+func isExcluded(s []string, str string) bool {
+	if s == nil {
+		return false
+	}
+	for _, v := range s {
+		if str == v {
+			return true
+		}
+	}
+	return false
+}
+
+// merge the spec fields of all MC manifests except the ones that are in the excluded list
+func MergeManifests(dataMap map[string]interface{}, excludes []string) (map[string]interface{}, error) {
+	// key is role, value is a list of MCs
+	configs := make(map[string][]*machineconfigv1.MachineConfig)
+
+	for k, v := range dataMap {
+		if isExcluded(excludes, k) {
+			continue
+		}
+		err := addMachineConfig(v, configs)
+		if err != nil {
+			return dataMap, err
+		}
+		// remove the individual file entries
+		delete(dataMap, k)
+	}
+
+	for k, v := range configs {
+		var osImageURL string = ""
+		//It only uses OSImageURL provided by the CVO
+		merged, err := mcfgctrlcommon.MergeMachineConfigs(v, osImageURL)
+		if err != nil {
+			return nil, err
+		}
+
+		merged.SetName(fmt.Sprintf("%s-%s", mcName, k))
+		merged.ObjectMeta.Labels = make(map[string]string)
+		merged.ObjectMeta.Labels[mcRoleKey] = k
+		merged.TypeMeta.APIVersion = machineconfigv1.GroupVersion.String()
+		merged.TypeMeta.Kind = "MachineConfig"
+
+		// Marshal to json
```

Comment:
Maybe a bit of a note would be good about why we want to marshal to json, then unmarshal from json?

Reply:
I added a note.